src/main/java/com/muxiu1997/sharewhereiam/integration/Mods.java

```diff
@@ -0,0 +1,27 @@
+package com.muxiu1997.sharewhereiam.integration;
+
+import cpw.mods.fml.common.Loader;
+
+public enum Mods {
+
+    VisualProspecting("visualprospecting"),
+    JourneyMap("journeymap"),
+    Navigator("navigator"),
+
+    ;
+
+    public final String modid;
+    private boolean loaded;
+
+    Mods(String modid) {
+        this.modid = modid;
+    }
+
+    public boolean isLoaded() {
+        return loaded;
+    }
+
+    public void check() {
+        this.loaded = Loader.isModLoaded(modid);
+    }
```

Comment:
Instead of a `check()` method that must be called at some point in loading, you can edit `isLoaded()` to look more like this:
```java
public boolean isLoaded() {
    if (loaded == null) {
        loaded = Loader.isModLoaded(modid);
    }
    return loaded;
}
```
And also changing `loaded` from `boolean` to `Boolean`